addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -0,0 +1,800 @@
+                    get: () => this.observable[this.__getStoredRelationName(relationName)],
+                });
+            }
+
+            // Class attribute access (redirect to observable)
+            const classAttributeNames = this._getListOfClassAttributeNames();
+            for (const classAttributeName of classAttributeNames) {
+                Object.defineProperty(this, classAttributeName, {
+                    get: () => this.observable[classAttributeName],
+                });
+            }
+
+            this.__init(this);
+            this.__update(this, data);
+        }
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * @static
+         * @returns {mail.messaging.entity.Entity[]} all instances of this entity
+         */
+        static get all() {
+            const { state } = this.env.store;
+            return Object.values(state.entities).filter(entity => entity instanceof this);
+        }
+
+        /**
+         * Single way to create new entity instances in the store.
+         *
+         * @static
+         * @param {Object} data data object with initial data, including relations.
+         * @returns {mail.messaging.entity.Entity} newly created entity
+         */
+        static create(data) {
+            const { state } = this.env.store;
+            const entity = new this(data);
+            Object.defineProperty(entity, 'env', {
+                get: () => this.env,
+            });
+            entity.localId = entity._createInstanceLocalId(data);
+            this.__init(entity);
+
+            state.entities[entity.localId] = entity;
+            // ensure observable version of entity is handled.
+            const proxifiedEntity = state.entities[entity.localId];
+            this.__update(proxifiedEntity, data);
+            return proxifiedEntity;
+        }
+
+        /**
+         * @static
+         */
+        static delete() {
+            for (const entity of this.all) {
+                entity.delete();
+            }
+            this.__delete(this);
+        }
+
+        /**
+         * @static
+         * @param {any} id
+         * @returns {mail.messaging.entity.Entity|undefined}
+         */
+        static fromId(id) {
+            const allEntities = this.all;
+            return allEntities.find(entity => entity.id === id);
+        }
+
+        /**
+         * @static
+         * @param {string|mail.messaging.entity.Entity|undefined} entityOrLocalId
+         * @returns {mail.messaging.entity.Entity|undefined}
+         */
+        static get(entityOrLocalId) {
+            const { state } = this.env.store;
+            if (entityOrLocalId === undefined) {
+                return undefined;
+            }
+            const entity = state.entities[
+                entityOrLocalId.isEntity
```

Comment:
For later: I don't think entity should be supported as a param of this method.
Another method for getting "refreshed" entity can be created if needed.

Reply:
The support of a local id or an entity is a consequence of supporting link/unlink with data being either local id(s) or entity(ies). For instance, we could either do `messaging.link({ discuss: discussEntity })` or `messaging.link({ discuss: discussLocalId })`, and both should work the same.